### PR TITLE
PEN-1594 Ability to use See More button when using Collections conten…

### PR DESCRIPTION
**Pull request by ktran-wizeline**
…t source

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
_Information about what you changed for this PR_
Rework on PEN-1594.

## Jira Ticket
- [PEN-1594](https://arcpublishing.atlassian.net/browse/PEN-1594)

Acceptance Criteria
copy from ticket
What should be happening
Clicking the read more button portion of the results-list block should load more stories when that block is configured with the content-api-collections content source.

Test Steps
1. Add test steps a reviewer must complete to test this PR
2. In Fusion-News-Theme repo, checkout master & git pull
3. After checking out this feature branch in fusion-news-theme-blocks, run rm -rf ./node_modules && npx lerna clean && npm i && npx lerna bootstrap && cd blocks/header-nav-chain-block && npm i && cd ../..
4. Run: npx fusion start -f -l @wpmedia/results-list-block
5. Create a page then add the feature results-list-block and configure content-api-collections for testing

### Before
_Example: When I clicked the search button, the button turned red._
The result list could not present more than 20 items

[include screenshot or gif or link to video, storybook would be awesome]

### After
_Example: When I clicked the search button, the button turned green._

Supported pagination on the result list, present more than 20 items

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working. 
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist 
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr. 
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/results-list-block/features/results-list/default.jsx [hunk 1]
```diff
@@ -128,39 +128,54 @@ class ResultsList extends Component {
...
-        query.size = '20';
+        from = parseInt(contentConfigValues.from, 10) || 0;
+        query.from = String(from + size + 1);
         this.fetchContent({
-          collection: {
+          seeMore: {
             source: contentService,
             query,
+            transform: (data) => {
+              if (!data || data.content_elements.length === 0) {
```
Comment (beltrancaliz) [inline blocks/results-list-block/features/results-list/default.jsx]: We should probably check that data has `content_elements` before  we attempt to get the length

File: blocks/results-list-block/features/results-list/default.jsx [hunk 2]
```diff
@@ -128,39 +128,54 @@ class ResultsList extends Component {
...
-      const { resultList, count } = this.state;
+      const { resultList } = this.state;
       this.state.storedList = resultList;
+
+      if (listContentConfig.contentService === 'content-api-collections') {
+        const query = { ...contentConfigValues };
+        const from = parseInt(contentConfigValues.from, 10) || 0;
+        const size = parseInt(contentConfigValues.size, 10) || 10;
+        query.from = from + size + 1;
+        console.log(from);
```
Comment (beltrancaliz) [inline blocks/results-list-block/features/results-list/default.jsx]:
```suggestion
```

Comment (ktran-wizeline) [reply blocks/results-list-block/features/results-list/default.jsx]: thank you, my bad :'(

File: blocks/results-list-block/features/results-list/default.jsx [hunk 3]
```diff
@@ -128,39 +128,53 @@ class ResultsList extends Component {
...
+      }
+      const { nextCollection } = this.state;
+
       // Check if there are available stories
       if (resultList?.content_elements) {
         // Hide button if no additional stories from initial content
         if ((resultList.content_elements.length >= resultList.count)
-          || (resultList.content_elements.length >= count)) {
+          || !nextCollection
+          || (nextCollection.content_elements.length === 0)) {
```
Comment (beltrancaliz) [inline blocks/results-list-block/features/results-list/default.jsx]: Sorry, missed this one on my last review. We should check the existence of `content_elements` here as well before checking its length

**Review by beltrancaliz — APPROVED** [review]

File: blocks/results-list-block/features/results-list/default.jsx [hunk 4]
```diff
@@ -50,13 +50,15 @@ class ResultsList extends Component {
...
       placeholderResizedImageOptions: {},
-      count: 0,
     };
     this.phrases = getTranslatedPhrases(getProperties(props.arcSite).locale || 'en');
-    this.fetchStories(false);
     this.fetchPlaceholder();
   }
 
+  componentDidMount() {
+    this.fetchStories(false);
```
Comment (ktran-wizeline) [inline blocks/results-list-block/features/results-list/default.jsx:59]: Hi @beltrancaliz I would request you review this PR again. I figured out some issue with content caching that causing the overlaps item issue with the last PR. I refactor something to get it works again. Thanks all for your help to review this!!!

**Review by beltrancaliz — APPROVED** [review]

**Pull request merged**